src/rdf-repository/grafter/rdf/repository.clj

```diff
@@ -135,13 +137,15 @@
     (.initialize)))
 
 (defn sparql-repo
-  "Given a query-url and an optional update-url String return a Sesame
-  SPARQLRepository for communicating with remote repositories."
+  "Given a query-url (String or IURI) and an optional update-url String
+  or IURI, return a Sesame SPARQLRepository for communicating with
+  remote repositories."
   ([query-url]
-     (doto (SPARQLRepository. query-url)
+     (doto (SPARQLRepository. (str query-url))
        (.initialize)))
   ([query-url update-url]
-     (doto (SPARQLRepository. query-url update-url)
+     (doto (SPARQLRepository. (str query-url)
+                              (str update-url))
```

Comment:
I think I actually suggested calling `str` on it too - to avoid the conditionals, and all `IURIs` should be able `str`ing into an actual URI... also if another object is passed that serialises in a meaningless way, I'm pretty sure the error message will be clear.